src/components/vm/overview/cpuModal.tsx

```diff
@@ -53,27 +56,39 @@ const dividers = (num) => {
     return divs;
 };
 
-const clamp = (value, max, min) => {
+const clamp = (value: number, max: number, min: number) => {
     return value < min || isNaN(value) ? min : (value > max ? max : value);
 };
 
-export const CPUModal = ({ vm, maxVcpu, models }) => {
+export const CPUModal = ({
+    vm,
+    maxVcpu,
+    models
+} : {
+    vm: VM,
+    maxVcpu: string,
+    models: string[],
+}) => {
     const Dialogs = useDialogs();
 
-    const [error, setError] = useState(undefined);
-    const [max, setMax] = useState(parseInt(vm.vcpus.max) || 1);
-    const [count, setCount] = useState(parseInt(vm.vcpus.count) || 1);
-    const [sockets, setSockets] = useState(vm.cpu.topology.sockets || 1);
-    const [threads, setThreads] = useState(vm.cpu.topology.threads || 1);
-    const [cores, setCores] = useState(vm.cpu.topology.cores || max);
+    function getInt(val: optString, def: number) {
+        return val ? parseInt(val) : def;
+    }
+
+    const [error, setError] = useState<undefined | { dialogError: string, dialogErrorDetail: string }>(undefined);
+    const [max, setMax] = useState(getInt(vm.vcpus.max, 1));
+    const [count, setCount] = useState(getInt(vm.vcpus.count, 1));
+    const [sockets, setSockets] = useState(getInt(vm.cpu.topology.sockets, 1));
+    const [threads, setThreads] = useState(getInt(vm.cpu.topology.threads, 1));
+    const [cores, setCores] = useState(getInt(vm.cpu.topology.cores, max));
     const [cpuMode, setCpuMode] = useState(vm.cpu.mode);
     const [cpuModel, setCpuModel] = useState(vm.cpu.model);
     const [isLoading, setIsLoading] = useState(false);
 
-    function onMaxChange(value) {
-        // Allow empty string
+    function onMaxChange(value: string) {
+        // Allow empty string, which we represent as zero.
         if (value == "") {
-            setMax("");
+            setMax(0);
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2185-af23a7d6-20250716-112918-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)